lib/nethash.h

```diff
@@ -0,0 +1,40 @@
+#ifndef LIB_NETHASH_H_
+#define LIB_NETHASH_H_
+
+/* Copyright 2013-present Barefoot Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Antonin Bas (antonin@barefootnetworks.com)
+ *
+ */
+
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+/// A collection of hashing functions commonly used in network protocols.
```

Comment:
Also would add a comment where this was adopted from:
https://github.com/p4lang/behavioral-model/blob/main/src/bm_sim/calculations.cpp